### PR TITLE
QT: add last block hash to debug ui

### PR DESCRIPTION
**Pull request by sc-9310**
Trivial addition to display last block hash next to last block time

image

### CODE REVIEW

**Comment by xdustinface** [comment]
Linter is unhappy, see 4566bdc9f47503e56c0e304a9053fc99b9241129

File: src/qt/forms/debugwindow.ui [hunk 1]
```diff
@@ -351,20 +351,34 @@
...
+         <property name="text">
+          <string>Last block hash</string>
+         </property>
+        </widget>
+       </item>
+       <item row="13" column="1">
+        <widget class="QLabel" name="lastBlockHash">
+         <property name="text">
+          <string>N/A</string>
+         </property>
```
Comment (UdjinM6) [inline src/qt/forms/debugwindow.ui:364]: Should probably make it selectable/copiable e.g.
```suggestion
         </property>
         <property name="cursor">
          <cursorShape>IBeamCursor</cursorShape>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
         <property name="textInteractionFlags">
          <set>Qt::LinksAccessibleByMouse|Qt::TextSelectableByKeyboard|Qt::TextSelectableByMouse</set>
         </property>
```

**Review by UdjinM6 — COMMENTED** [review]
Pls see inline suggestions

File: src/qt/clientmodel.cpp [hunk 2]
```diff
@@ -344,6 +354,7 @@ static void BlockTipChanged(ClientModel *clientmodel, bool initialSync, const CB
         QMetaObject::invokeMethod(clientmodel, "numBlocksChanged", Qt::QueuedConnection,
                                   Q_ARG(int, pIndex->nHeight),
                                   Q_ARG(QDateTime, QDateTime::fromTime_t(pIndex->GetBlockTime())),
+                                  Q_ARG(QString, clientmodel->getLastBlockHash()),
```
[the remark this answers is not in the document]
Comment (sc-9310) [reply src/qt/clientmodel.cpp]: Works both, but ye, switched

File: src/qt/forms/debugwindow.ui (hunk 1 above)
Comment (sc-9310) [reply src/qt/forms/debugwindow.ui:367]: Agree, makes sense, thanks.
Just moved up cursor prop above text to keep it similar

**Comment by UdjinM6** [comment]
Linter still complains, pls apply 1532975fc3

**Review by UdjinM6 — APPROVED** [review]
ACK

**Review by xdustinface — APPROVED** [review]
utACK

**Review by PastaPastaPasta — APPROVED** [review]
ACK, tested. Code looks good

File: src/qt/clientmodel.cpp [hunk 3]
```diff
@@ -153,6 +153,16 @@ QDateTime ClientModel::getLastBlockDate() const
     return QDateTime::fromTime_t(Params().GenesisBlock().GetBlockTime()); // Genesis block's time of current network
 }
 
+QString ClientModel::getLastBlockHash() const
+{
+    LOCK(cs_main);
+
+    if (chainActive.Tip())
+        return QString::fromStdString(chainActive.Tip()->GetBlockHash().ToString());
```
Comment (PastaPastaPasta) [inline src/qt/clientmodel.cpp:161]:
```suggestion
    if (chainActive.Tip()) return QString::fromStdString(chainActive.Tip()->GetBlockHash().ToString());
```

**Review by PastaPastaPasta — COMMENTED** [review]
% 1 nit

File: src/qt/clientmodel.cpp (hunk 3 above)
Comment (UdjinM6) [reply src/qt/clientmodel.cpp:161]: IMO it's ok-ish since it's kind of a copy-paste like thing of the function above.

**Pull request merged**